axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/StorageTier.java

```diff
@@ -191,14 +191,16 @@ public Long findNextSegment() {
                 long minTimestamp = System.currentTimeMillis() - retentionTimeInMillis;
                 long candidate = segmentIterator.next();
 
-
                 if (segments.firstKey() == candidate) {
                     return null;
                 }
 
-
-                Map.Entry<Long, Integer> firstEntry = segments.firstEntry();
-                if (dataFileResolver.apply(new FileVersion(firstEntry.getKey(), firstEntry.getValue())).lastModified() < minTimestamp) {
+                Integer candidateVersion = segments.get(candidate);
+                if (candidateVersion == null) {
```

Comment:
why candidateVersion can be null?
why should we return null if candidateVersion is null?